woocommerce/woocommerce
# Enhance Customer Email Password Reset include Login Username in Reset Password Link

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

This PR adds the username to the reset password link in WooCommerce. This enhancement is necessary for environments that utilize the REST API or GraphQL and require the username instead of the user ID for the password reset process.

### How to test the changes in this Pull Request:
Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Navigate to the Password Reset Page:
    - Go to `{{frontend url}}/my-account/lost-password/`

2. Request a Password Reset:
    - Enter a valid username or email address in the provided field.
    - Submit the request.

3. Verify Email Receipt:
    - Check your email inbox for a password reset email.
    - Confirm that the email has been received and contains a password reset link.

4. Inspect the Reset Link:
    - Ensure that the link in the email includes the username or email as a query parameter.
    - Verify that the link is correctly formatted and does not contain any errors.

5. Test the Password Reset Link:
    - Click on the reset link to be redirected to the password reset page.

6. Reset the Password:
    - Enter a new password in the provided fields.
    - Submit the new password.

7. Verify Password Update:
    - Confirm that the new password is accepted and the reset is successful.
    - Check for any error messages during the process.

8. Test Reset Link Without Username:
    - Use the password reset link from the email and manually remove the username query parameter.
    - Verify that you can still reset the password successfully and check for any errors or issues.









<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message Add username in email reset-password link

</details>

<details>

<summary>Changelog Entry Comment</summary>
This PR enhances the password reset functionality by including the username in the reset password link, to be able to use from

</details>

Thank you for reviewing this PR! Please let me know if you need any further information or adjustments.
